src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/personalrelationships/profiledetails/ContactPersonProfileDetailsReconciliationService.kt

```diff
@@ -82,12 +88,16 @@ class ContactPersonProfileDetailsReconciliationService(
   suspend fun CoroutineScope.checkPrisoners(
     ids: ReceiveChannel<PrisonerIds>,
     mismatches: Channel<String>,
-  ) = launch {
+  ): List<Job> {
+    val jobs = mutableListOf<Job>()
     for (id in ids) {
-      checkPrisoner(id.offenderNo)
-        ?.also { mismatches.send(it) }
+      jobs += launch {
+        checkPrisoner(id.offenderNo)
+          ?.also { mismatches.send(it) }
+      }
     }
-  }.join()
```

Comment:
This was the bug - the job launched to perform the mismatch check was joining on itself, hence blocking.